qiskit/ignis/verification/entanglement/parallelize.py

```diff
@@ -0,0 +1,449 @@
+
+        for x in range(length):
+            tier[x] = [[]]
+
+        parentlist = []
+        parentlist.append(parent)
+        ii = 0
+        while True:
+            totalchildren = []
+            for parent in parentlist:
+                children = self.nodes[parent]
+                if self.indicator:
+                    children = self.child_sorter(children, parent)
+                totalchildren += children
+                children = [a for a in children if a not in trashlist]
+                j = tierDM[parent]
+                for i, _ in enumerate(children):
+                    tier[j] += [[parent, children[i]]]
+                    tierDM[children[i]] = j+1
+                    j += 1
+                    trashlist.append(children[i])
+                parentlist = totalchildren
+
+            if len(trashlist) == length:
+                break
+            ii += 1
+            if ii > 50:
+                break
+
+        newtier = {}
+        for a in tier:
+            if [] in tier[a]:
+                tier[a].remove([])
+        for a in tier:
+            if tier[a] != []:
+                newtier[a] = tier[a]
+        tier = newtier
+
+        return tier
+
+    def get_ghz_layout(self, n, transpiled=True, barriered=True):
+        '''
+        Feeds the Tier Dict of the backend to create a basic
+        qiskit GHZ circuit with no measurement;
+        Args:
+           n: number of qubits
+           transpiled: toggle on/off transpilation - useful for tomography
+           barriered: yes/no whether to barrier each step of CNOT gates
+        Returns:
+           A GHZ Circuit
+       '''
+
+        tierdict = self.get_tier_dict()
+        q = QuantumRegister(n, 'q')
+        circ = QuantumCircuit(q)
+        circ.h(q[0])
+        trashlist = []
+        initial_layout = {}
+        for a in tierdict:
+            for aa in tierdict[a]:
+                if aa[0] not in trashlist:
+                    trashlist.append(aa[0])
+                    trashindex = trashlist.index(aa[0])
+                    initial_layout[q[trashindex]] = aa[0]
+                else:
+                    pass
+                if aa[1] not in trashlist:
+                    trashlist.append(aa[1])
+                    trashindex = trashlist.index(aa[1])
+                    initial_layout[q[trashindex]] = aa[1]
+                else:
+                    pass
+                circ.cx(q[trashlist.index(aa[0])],
+                        q[trashlist.index(aa[1])])
+
+                if len(trashlist) == n:
+                    break
+            if barriered:
+                circ.barrier()
+            if len(trashlist) == n:
+                break
+
+        if transpiled:
+            circ = qiskit.compiler.transpile(circ, backend=self.backend,
+                                             initial_layout=initial_layout)
+
+        return circ, initial_layout
+
+    def get_measurement_circ(self, n, extent):
```

Comment:
There seems to be code duplication between here and linear.py, unless I'm missing subtle differences. It might be best to unify a little.

Reply:
Yes, this is noted. In some of the cases implementation is almost the same, but in others very different. The interfaces are the same